internal/http.go

```diff
@@ -0,0 +1,239 @@
+package internal
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/operate-first/curator-operator/pkg/signals"
+	"github.com/spf13/viper"
+
+	"github.com/jackc/pgx/v4"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// Server defines the http server configuration
+type Server struct {
+	Addr        string
+	ErrorLog    *log.Logger
+	BaseContext func(net.Listener) context.Context
+}
+
+// HTTPError defines the http error codes
+type HTTPError struct {
+	code    int
+	message string
+}
+
+func InitializeRouter(ctx context.Context) {
+	var reportLog = ctrl.Log.WithName("report")
+	http.HandleFunc("/download", downloadReport)
+
+	// viper.SetDefault("HTTP.PORT", ":5000")
+	httpServerPort := viper.Get("HTTP.PORT").(string)
```

Comment:
This needs to be handled at the main package level. We can use the cobra package to get a CLI flag (and default to a port value) there, and pass that value to this constructor function.